runtime/lua/vim/lsp/util.lua

```diff
@@ -226,6 +226,142 @@ end
 -- function M.glob_to_regex(glob)
 -- end
 
+function table.slice(tbl, first, last)
+  local sliced = {}
+
+  for i = first or 1, last or #tbl do
+    sliced[#sliced+1] = tbl[i]
+  end
+
+  return sliced
+end
+
+local function first_difference(old_buffer, new_buffer)
+	local line_count = math.min(#old_buffer, #new_buffer)
+	if line_count == 0 then return { 0, 0} end
+	local start_line_idx = line_count
+	for i = 1, line_count do
+		if old_buffer[i] ~= new_buffer[i] then
+			start_line_idx = i
+			break
+		end
+	end
+	if start_line_idx >= line_count then
+		return line_count, #old_buffer[line_count]
```

Comment:
What if the change happens on line `line_count` but before the last char?

Perhaps `start_line_idx` should be initialized with `line_count+1` or another value not in the range 1..`line_count`.

Reply:
Yes, agreed.